cmd/oceanbench/api.go

```diff
@@ -91,6 +91,40 @@ func apiHandler(w http.ResponseWriter, r *http.Request) {
 			fmt.Fprint(w, string(enc))
 			return
 
+		case "devices":
+			switch val {
+			case "site":
+				// Check that the user has at least write access.
+				if len(strings.Split(p.Data, ":")) != 2 {
+					fmt.Fprint(w, "no site data in profile")
```

Comment:
Perhaps use writeHttpError instead of just writing a string as a response. Here and in all places an error is sent back below.